app/services/mapping/governmentagencygoodsitem/AdditionalDocumentsBuilder.scala

```diff
@@ -35,13 +46,38 @@ import wco.datamodel.wco.declaration_ds.dms._2.{
 import scala.collection.JavaConverters._
 
 object AdditionalDocumentsBuilder {
+  def buildThenAdd(
+    exportItem: ExportItem,
+    wcoGovernmentAgencyGoodsItem: GoodsShipment.GovernmentAgencyGoodsItem
+  ): Unit =
+    exportItem.documentsProducedData.foreach { documentsProduced =>
+      documentsProduced.documents.map(createGoodsItemAdditionalDocument).foreach { goodsItemAdditionalDocument =>
+        wcoGovernmentAgencyGoodsItem.getAdditionalDocument.add(createAdditionalDocument(goodsItemAdditionalDocument))
+      }
+    }
 
   def build(procedureCodes: Seq[GovernmentAgencyGoodsItemAdditionalDocument]): java.util.List[AdditionalDocument] =
     procedureCodes
       .map(document => createAdditionalDocument(document))
       .toList
       .asJava
 
+  //TODO get rid of the interim model GovernmentAgencyGoodsItemAdditionalDocument and map to wco dec directly
```

Comment:
Just checking if you still need to keep this comment.

Reply:
yes as this needs to go